### PR TITLE
Remove any type where possible. Prefer TS type system.

### PR DESCRIPTION
**Pull request by IEvangelist**

### CODE REVIEW

File: vscode-dotnet-runtime-library/src/Utils/WebRequestWorker.ts [hunk 1]
```diff
@@ -9,15 +9,15 @@ import { IEventStream } from '../EventStream/EventStream';
...
+    private cachedData: string | undefined;
+    private currentRequest: Promise<string | undefined> | undefined;
 
     constructor(private readonly extensionState: Memento,
                 private readonly eventStream: IEventStream,
                 private readonly uri: string,
                 private readonly extensionStateKey: string) {}
 
-    public async getCachedData(): Promise<string> {
+    public async getCachedData(): Promise<string | undefined> {
```
Comment (sfoslund) [inline vscode-dotnet-runtime-library/src/Utils/WebRequestWorker.ts:20]: Why is adding the undefined here necessary?

Comment (IEvangelist) [reply vscode-dotnet-runtime-library/src/Utils/WebRequestWorker.ts:20]: The `this.extensionState.get<string>` returns the union `string | undefined`, making `this.cachedData` a union of `string | undefined`, which then makes the entire method that too.

File: vscode-dotnet-runtime-library/src/Acquisition/VersionResolver.ts [hunk 2]
```diff
@@ -24,7 +24,7 @@ export class VersionResolver implements IVersionResolver {
     public async getFullVersion(version: string): Promise<string> {
         try {
             const response = await this.webWorker.getCachedData();
-            const releasesVersions = new ReleasesResult(response);
+            const releasesVersions = new ReleasesResult(response!);
```
[the remark this answers is not in the document]
Comment (IEvangelist) [reply vscode-dotnet-runtime-library/src/Acquisition/VersionResolver.ts:27]: I'm good with that. I didn't really know what y'all expected here.

**Comment by wli3** [comment]
Although is there any lint rule we could set?

**Comment by sfoslund** [comment]
@wli3 Sure, I'll add that change to #74 (since I don't seem to have permission to push to this branch 😄)

**Pull request merged**